Add ability for course staff to always view the results.

mtyaka opened the pull request:
Course staff users can now view the results without voting.
The can also see the results for polls/surveys marked as 'private'.

If user has sufficient permissions, a 'View results' link is shown on the bottom of the component in the LMS.

mtyaka committed:
Simplify test setup and set up travis CI.

No need to manually install xblock-sdk & dependencies anymore.
Inside a fresh virtual environment, run `sh install_test_deps.sh`,
then run the tests with `python run_tests.py`. That's it.

mtyaka committed:
Add ability for staff to always view the results.

Course staff users can now view the results without voting.
The can also see the results for polls/surveys marked as 'private'.

If user has sufficient permissions, a 'View results' link is shown
on the bottom of the component in the LMS.

mtyaka committed:
Tweaks for pep8/pylint.

Kelketek commented on pylintrc:
```diff
@@ -0,0 +1,27 @@
+	unused-argument
+
+[SIMILARITIES]
+min-similarity-lines=8
```
Nit: Missing a newline here.

Kelketek commented:
You're going to have to rebase this again after I address the notes you gave me. But as long as you do that and the tests pass, :+1: LGTM

mtyaka committed:
Add ability to grant view results permission to additional groups.

You can now use the XBLOCK_POLL_EXTRA_VIEW_GROUPS django setting
to allow users from additional groups to view the results.

By default only course staff has permission to view the results.

antoviaque committed:
Merge pull request #6 from mckinseyacademy/max_attempts

Max Submissions

Kelketek commented on poll/poll.py:
```diff
@@ -166,6 +176,25 @@ def can_vote(self):
+            else:
+                # Check if user is member of a group that is explicitly granted
+                # permission to view the results through django configuration.
+                group_names = getattr(settings, 'XBLOCK_POLL_ADDITONAL_GROUPS_WITH_VIEW_RESULTS_PERMISSION', [])
```
Good golly. Can we call this something less verbose? Maybe `XBLOCK_POLL_EXTRA_VIEW_GROUPS`?

mtyaka replied:
@Kelketek lol, thanks for catching this, I'm not sure what I was thinking.

Kelketek commented:
@mtyaka Aside from changing the name of the setting (Don't forget to do it in the README as well), this seems good to go.

Kelketek commented:
Do note, however, you'll have to close this PR and re-open it against master.

Kelketek committed:
Addressed nits from max_submissions task.

Kelketek committed:
Added counter display for polls with maximum submissions greater than 1.

Kelketek committed:
Fixed Javascript scoping issues and missing return statement in vote.

mtyaka commented:
Closed this and created a PR against master at https://github.com/mckinseyacademy/xblock-poll/pull/10
